enhancements/console/customize-perspectives.md

```diff
@@ -0,0 +1,278 @@
+- Provide a way that cluster admins can do this based on resource permissions (RBAC).
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+We can extend the existing `operator.openshift.io/v1` / `Console` CRD. 
+We can add 
+`spec.customization.perspectives` which can have list of customization details.
+
+### Workflow Description
+
+To disable perspective for users with limited permissions we need a list of options with a perspective id and the required permissions.
+
+```text
+├── Spec
+│   ├── customization
+|       ├── perspectives 
+|           ├── id
+|           ├── requiresAccessReview
+|               ├── kind
+|               ├── verb
+└── ...
+```
+
+This might be extendable in the future if we want replace (hide one, show another) a perspective based on a resource permissions:
+
+```text
+├── Spec
+│   ├── customization
+|       ├── perspectives 
+|           ├── id
+|           ├── requiresAccessReview
+|           |   ├── kind
+|           |   ├── verb
+|           ├── missingAccessReview
+|               ├── kind
+|               ├── verb
+└── ...
+```
+
+Each perspective conforms to the following schema:
+
+- `id: string` , defines the id of perspective to disable. Incorrect or unknown ids will be ignored.
+
+// +optional
+- `requiresAccessReview: authV1.ResourceAttributes[]` , defines a list of permission checks. The perspective will only be shown when all checks are successful.
+
+// +optional
+- `missingAccessReview: authV1.ResourceAttributes[]` , defines a list of permission checks. The perspective will only be shown when at least one check fails.
+
+### API Extensions
+
+Extend the existing `operator.openshift.io/v1` / `Console` CRD to add `spec.customization.perspectives` which can have list of customization details.
+
+So the final yaml should look like this one:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata: 
+  name: cluster
+  ...
+spec:
+  customization:
+    perspectives:
+      - id: admin
+        requiresAccessReview:
+          - kind: namespaces
+            verb: list
+      - id: dev
+        state: Enabled/Disabled
+```
+
+Example yaml based on resource permission:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata: 
+  name: cluster
+  ...
+spec:
+  customization:
+    perspectives:
+      - id: admin
+        requiresAccessReview:
+          - kind: namespaces
+            verb: list
+      - id: minimal-custom-admin
+        missingAccessReview:
+          - kind: namespaces
+            verb: list
+```
+### Risks and Mitigations
+
+**Default perspective**: When all the perspectives are hidden from a user or for all users, we always show Admin perspective by default and if admin has to 
+revert back the changes, admin will update the yaml by removing specific perspective details. 
+
+### Drawbacks
+
+N/A
+
+## Design Details
+
+N/A
+
+### Test Plan
+
+Testing will be carried out with the usual Console unit and e2e test suites.
```

Comment:
Are there any specific test cases you can think of that you might want QE to run through, in particular I'm wondering about some of the cases mentioned above:
- What happens if the user doesn't have permissions
- What happens when all perspectives are hidden
- What happens if a plugin adds a perspective